limitador-server/src/main.rs

```diff
@@ -251,16 +252,16 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
                     KeyValue::new("service.name", "limitador"),
                 ])))
                 .install_batch(opentelemetry_sdk::runtime::Tokio)?;
+            let tracing_level = level.max(LevelFilter::INFO);
             let telemetry_layer = tracing_opentelemetry::layer().with_tracer(tracer);
             tracing_subscriber::registry()
-                .with(level)
                 .with(metrics_layer)
                 .with(fmt_layer)
+                .with(tracing_level)
                 .with(telemetry_layer)
                 .init();
         } else {
             tracing_subscriber::registry()
-                .with(level)
```

Comment:
Worth mentioning that removing this global level filter means all spans hit the metrics layer regardless of the configured verbosity now